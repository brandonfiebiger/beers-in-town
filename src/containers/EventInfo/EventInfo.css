img {
  border-radius: 190px;
  width: 39vw;
}

.event-info {
  background: rgba(25, 25, 25, .5);
  color: #f9f9f9;
  width: 50vw;
  margin: 0 auto;
  border-radius: 20px;
  height: 100vh;
  overflow: scroll;
  padding: 3rem;
  margin-top: 2rem;
}

a {
  color: white;
}