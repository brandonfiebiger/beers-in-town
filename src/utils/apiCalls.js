import { apikey } from './variables';


export const fetchBreweryDataByLocation = async (latitude, longitude) => {
  const response = await fetch(`https://api.meetup.com/find/venues?key=${apikey}&sign=true&photo-host=secure&text=brewery&lon=${longitude}&radius=100&lat=${latitude}&page=100`);
  const breweries = await response.json();
  return breweries;
}

export const fetchEventDataByLocation = async (latitude, longitude) => {
  const response = await fetch(`https://api.meetup.com/find/upcoming_events?key=${apikey}&sign=true&photo-host=public&lon=${longitude}&page=50&text=craft beer, brewery&radius=50&lat=${latitude}`);
  const breweries = await response.json();
  return Promise.all(breweries.events);
}
