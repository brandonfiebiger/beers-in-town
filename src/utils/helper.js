export const eventCleaner = (events) => {
  return events.map(event => {
    // console.log(event);
    let desc;
    const { description, venue, group } = event
    desc = description
    if(!description) {
      desc = '<p>no description available</p>'
    }
    return {desc, groupName: event.group.name, time: event.local_time, date: event.local_date, venue, group, name: event.name, id: event.id }
  })
}