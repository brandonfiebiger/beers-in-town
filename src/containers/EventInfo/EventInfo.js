import React from 'react';
import ReactHtmlParser, { processNodes, convertNodeToElement, htmlparser2 } from 'react-html-parser';
import PropTypes from 'prop-types';
import './EventInfo.css';


export const EventInfo = (props) => {

  return(
    <div className="event-info">
      <h1>{ props.eventProps.name }</h1>
      <h2>By: { props.eventProps.groupName }</h2>
      { props.eventProps.ven ? <h3>At: { props.eventProps.ven.name }</h3> : '' }
      <p>{ props.eventProps.date }</p>
      <p>{ props.eventProps.time }</p>
      <section>{ ReactHtmlParser(props.eventProps.desc) }</section>
    </div>
  )
}

export default EventInfo;

EventInfo.propTypes = {
  eventProps: PropTypes.object
};