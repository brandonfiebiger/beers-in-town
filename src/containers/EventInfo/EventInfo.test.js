import React from 'react';
import { EventInfo } from './EventInfo';
import { shallow, mount } from 'enzyme';
import { mockEventProps, mockEventPropsWithoutVenue } from '../../utils/mockData';


describe('EventInfo', () => {

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<EventInfo eventProps={mockEventProps}/>)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should match the snapshot without a venue', () => {
    wrapper = shallow(<EventInfo eventProps={mockEventPropsWithoutVenue} />)

    expect(wrapper).toMatchSnapshot()
  })
})