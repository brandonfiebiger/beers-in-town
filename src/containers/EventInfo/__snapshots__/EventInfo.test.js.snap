// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventInfo should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    eventProps={
        Object {
            "date": "Tomorrow",
            "desc": "Really good",
            "groupName": "the cool people",
            "name": "Brandon",
            "time": 4,
            "ven": Object {
              "name": "cool spot",
            },
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Brandon
</h1>,
        <h2>
          By: 
          the cool people
</h2>,
        <h3>
          At: 
          cool spot
</h3>,
        <p>
          Tomorrow
</p>,
        <p>
          4
</p>,
        <section>
          Really good
</section>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Brandon",
        },
        "ref": null,
        "rendered": "Brandon",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "By: ",
            "the cool people",
          ],
        },
        "ref": null,
        "rendered": Array [
          "By: ",
          "the cool people",
        ],
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "At: ",
            "cool spot",
          ],
        },
        "ref": null,
        "rendered": Array [
          "At: ",
          "cool spot",
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Tomorrow",
        },
        "ref": null,
        "rendered": "Tomorrow",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 4,
        },
        "ref": null,
        "rendered": 4,
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Really good",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Really good",
        ],
        "type": "section",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Brandon
</h1>,
          <h2>
            By: 
            the cool people
</h2>,
          <h3>
            At: 
            cool spot
</h3>,
          <p>
            Tomorrow
</p>,
          <p>
            4
</p>,
          <section>
            Really good
</section>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Brandon",
          },
          "ref": null,
          "rendered": "Brandon",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "By: ",
              "the cool people",
            ],
          },
          "ref": null,
          "rendered": Array [
            "By: ",
            "the cool people",
          ],
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "At: ",
              "cool spot",
            ],
          },
          "ref": null,
          "rendered": Array [
            "At: ",
            "cool spot",
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Tomorrow",
          },
          "ref": null,
          "rendered": "Tomorrow",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 4,
          },
          "ref": null,
          "rendered": 4,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Really good",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Really good",
          ],
          "type": "section",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`EventInfo should match the snapshot without a venue 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    eventProps={
        Object {
            "date": "Tomorrow",
            "desc": "Really good",
            "groupName": "the cool people",
            "name": "Brandon",
            "time": 4,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Brandon
</h1>,
        <h2>
          By: 
          the cool people
</h2>,
        "",
        <p>
          Tomorrow
</p>,
        <p>
          4
</p>,
        <section>
          Really good
</section>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Brandon",
        },
        "ref": null,
        "rendered": "Brandon",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "By: ",
            "the cool people",
          ],
        },
        "ref": null,
        "rendered": Array [
          "By: ",
          "the cool people",
        ],
        "type": "h2",
      },
      "",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Tomorrow",
        },
        "ref": null,
        "rendered": "Tomorrow",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 4,
        },
        "ref": null,
        "rendered": 4,
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Really good",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Really good",
        ],
        "type": "section",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Brandon
</h1>,
          <h2>
            By: 
            the cool people
</h2>,
          "",
          <p>
            Tomorrow
</p>,
          <p>
            4
</p>,
          <section>
            Really good
</section>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Brandon",
          },
          "ref": null,
          "rendered": "Brandon",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "By: ",
              "the cool people",
            ],
          },
          "ref": null,
          "rendered": Array [
            "By: ",
            "the cool people",
          ],
          "type": "h2",
        },
        "",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Tomorrow",
          },
          "ref": null,
          "rendered": "Tomorrow",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 4,
          },
          "ref": null,
          "rendered": 4,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Really good",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Really good",
          ],
          "type": "section",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
